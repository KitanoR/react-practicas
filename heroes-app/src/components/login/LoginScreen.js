import React from 'react'

export const LoginScreen = ({ history }) => {
    const handleLogin = () => {
        history.replace('/marvel');
    }
    return (
        <div className="container mt-5">
            <h1>Login screen</h1>
            <hr />
            <p>
                Ingresa 
            </p>
            <button
                className="btn btn-primary"
                onClick={handleLogin}
            >
                Login
            </button>
        </div>
    )
}
