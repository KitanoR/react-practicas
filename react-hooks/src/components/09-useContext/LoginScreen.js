import React, { useContext } from 'react'
import { UserContext } from './UserContext';

export const LoginScreen = () => {
    const { user, setUser } = useContext(UserContext);
    return (
        <div>
            <h1>Login Screen</h1>
            <hr />
            <button
                className="btn btn-primary"
                onClick={() => setUser({ id: 1, name: 'cayetano', age: 23 })}
            >
                Login
            </button>
        </div>
    )
}
