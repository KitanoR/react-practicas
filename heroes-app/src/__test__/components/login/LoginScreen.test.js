import { mount } from 'enzyme';
import { AuthContext } from '../../../auth/AuthContext';
import { LoginScreen } from '../../../components/login/LoginScreen';
import { types } from '../../../types/types';

describe('Pruebas en <LoginScreen />', () => {
    const historyMock = {
        length: 10, 
        goBack: jest.fn(),
        push: jest.fn(),
        replace: jest.fn(),
    };
    const contextValue = {
        dispatch: jest.fn(),
        user: {
            logged: false,
        }
    }
    const wrapper = mount(
        <AuthContext.Provider value={contextValue}>
            <LoginScreen history={historyMock} />
        </AuthContext.Provider>
    );
    test('should de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de realizar el dispatch y la navegacion', () => {
        const handleClick = wrapper.find('button').prop('onClick');
        handleClick();
        expect(contextValue.dispatch).toHaveBeenCalledWith({
            type: types.login,
            payload: {
                name: 'cayetano'
            }
        });
        expect(historyMock.replace).toHaveBeenCalledWith('/');
        localStorage.setItem('lastPath', '/dc');
        handleClick();
        expect(historyMock.replace).toHaveBeenCalledWith('/dc');

    })
    
})
