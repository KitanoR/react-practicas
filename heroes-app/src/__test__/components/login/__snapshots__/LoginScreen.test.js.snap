// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <LoginScreen /> should de mostrarse correctamente 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login screen
  </h1>
  <hr />
  <p>
    Ingresa
  </p>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
